lib/Parse/ParseDecl.cpp

```diff
@@ -5621,8 +5621,56 @@ void Parser::parseAbstractFunctionBody(AbstractFunctionDecl *AFD) {
     Context.Stats->getFrontendCounters().NumFunctionsParsed++;
 
   ParserResult<BraceStmt> Body = parseBraceItemList(diag::invalid_diagnostic);
-  if (!Body.isNull())
-    AFD->setBody(Body.get());
+  if (!Body.isNull()) {
+    // If the body consists of a single expression, turn it into a return 
+    // statement.
+    BraceStmt * BS = Body.get();
+    AFD->setBody(BS);
+    // FIXME: FIXME_NOW: DETERMINE: Do we need to handle the code completion 
+    //        case here?  It is being handled for closure 
+    //        (Parser::parseExprClosure).
+    if (BS->getNumElements() == 1) {
+      auto Element = BS->getElement(0);
+      // FIXME: FIXME_NOW: DETERMINE: Do we need to do this?  It's being done
+      //        for closures (Parser::parseExprClosure) but it may not be 
+      //        relevant here.
```

Comment:
I think it's not currently relevant because we don't infer return types, but it would still be nice to be uniform.

Reply:
Okay, I uncommented and cleaned this up in 
https://github.com/apple/swift/pull/23251/commits/a310279dabda30f1de50b2a6bb2c11c0ef20868f .